test/unit/classification_test.rb

```diff
@@ -593,6 +595,43 @@ def setup
     assert_equal [3,4], classification.enc['base'][key.key]
   end
 
+  test 'enc should return correct values for multi-key matchers' do
+
+    key = FactoryGirl.create(:lookup_key, :as_smart_class_param, :use_puppet_default => true,
+                             :override => true, :key_type => 'string', :merge_overrides => false,
+                             :path => "hostgroup,organization\nlocation",
+                             :puppetclass => puppetclasses(:two))
+
+    parent_hostgroup = FactoryGirl.create(:hostgroup,
+                                          :puppetclasses => [puppetclasses(:two)],
+                                          :environment => environments(:production))
+    hostgroup = classification.send(:hostgroup)
+    hostgroup.update_attributes(:parent => parent_hostgroup)
+
+    as_admin do
+      LookupValue.create! :lookup_key_id => key.id,
```

Comment:
@eLobato I don't really understand what's unneeded. I can try to move this to FactoryGirl but the only associations I see are to lookup_key which is needed and to puppetclasses and environments so the classification will find these lookup_values and use them.